frank-flow/src/frontend/src/app/flow/canvas/canvas.component.ts

```diff
@@ -92,13 +102,28 @@ export class CanvasComponent implements AfterViewInit, OnDestroy {
           this.errors = currentFile.errors;
           this.locked = this.XmlErrorsFound();
           if (currentFile.flowStructure && currentFile.flowNeedsUpdate) {
+            this.currentFile = currentFile;
```

Comment:
The currentFile needs to be set every time next is called, but doesn't always need to be generated.
In case a user drags a node, the flow shouldn't be updated because the user places the node there already. But it should update the currentFile.
This needs to be done because the settings observer will use the currentFile member. This should be the newest currentFile.

Reply:
Fixed all the bugs